src/sync/push.rs

```diff
@@ -116,6 +119,79 @@ pub enum PushError {
     SerializePushError(serde_json::error::Error),
 }
 
+/// Find pending commits between the base snapshot and the main head and push
```

Comment:
in the last pr this comment should have gone inside the function, not on it. maybe move it to above line 134 (let dag_read =) below?